assets/cases/ui/21.video-player/video-player-ctrl.ts

```diff
@@ -65,32 +62,22 @@ export class VideoPlayerCtrl extends Component {
 
     onPlayLocalVideo () {
         this.videoPlayer.resourceType = VideoPlayer.ResourceType.LOCAL;
-        if (this.videoPlayer.clip === this.videClip) {
-            this.videoPlayer.play();
-        }
-        else {
+        if (this.videoPlayer.clip !== this.videClip){
             this.videoPlayer.clip = this.videClip;
         }
+        if (!this.videoPlayer.isPlaying) {
+            this.videoPlayer.play();
+        }
```

Comment:
Why need to check the status before playing? VideoPlayer will not check the status?

Reply:
Ok, it may cause to replay.

Reply:
Currently, we do not do this check in VideoPlayer. And different platform have different strategies when repeatly call play. So I think it's a foolproof way to check in script.